components/AnimatedLoading.js

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { Animated } from 'react-native';
+import LottieView from 'lottie-react-native';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { View } from '@shoutem/ui';
+import { animations } from '../assets';
+import { usePreviousValue } from '../hooks';
+
+const AnimatedView = Animated.createAnimatedComponent(View);
+
+function AnimatedLoading({
+  children,
+  finishAnimationCallback,
+  loading,
+  style,
+}) {
+  const prevLoading = usePreviousValue(loading);
+  const animateProgress = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
```

Comment:
do you really need this ? useEffect triggers only on the control prop change. Meaning that you could just check if the loading is true when it triggers. If it is, it means that it changed from false to true, which is the step you are monitoring. Like this : 

> useEffect(() => {
>   if (loading) {
>     startAnimation();
>   }
> }, [startAnimation, loading])

Reply:
I know, I've done it like that in first place. I've explained why I've implemented this in PR description :D 

> created usePreviousValue hook. We will use it to determine when some prop changed when compared to previous value, like we did in componentDidUpdate. This component probably shouldn't take care of this - component rendering AnimatedLoading should be sending proper loading states and not re-render AnimatedLoading, but it's a failsafe for when things don't work properly in parent component. This operations makes sure loading animation animates properly, at almost no-cost